cmake/modules/FindFFMPEG.cmake

```diff
@@ -165,153 +265,44 @@ if(NOT ENABLE_INTERNAL_FFMPEG OR KODI_DEPENDSBUILD)
 
-  else()
-    message(STATUS "FFmpeg ${REQUIRED_FFMPEG_VERSION} not found, falling back to internal build")
-    unset(FFMPEG_INCLUDE_DIRS)
-    unset(FFMPEG_INCLUDE_DIRS CACHE)
-    unset(FFMPEG_LIBRARIES)
-    unset(FFMPEG_LIBRARIES CACHE)
-    unset(FFMPEG_DEFINITIONS)
-    unset(FFMPEG_DEFINITIONS CACHE)
-  endif()
-
-  if(FFMPEG_FOUND)
-    set(FFMPEG_LDFLAGS ${PC_FFMPEG_LDFLAGS} CACHE STRING "ffmpeg linker flags")
-
-    # check if ffmpeg libs are statically linked
-    set(FFMPEG_LIB_TYPE SHARED)
-    foreach(_fflib IN LISTS FFMPEG_LIBRARIES)
-      if(${_fflib} MATCHES ".+\.a$" AND PC_FFMPEG_STATIC_LDFLAGS)
-        set(FFMPEG_LDFLAGS ${PC_FFMPEG_STATIC_LDFLAGS} CACHE STRING "ffmpeg linker flags" FORCE)
-        set(FFMPEG_LIB_TYPE STATIC)
-        break()
-      endif()
-    endforeach()
-
-    set(FFMPEG_LIBRARIES ${FFMPEG_LIBAVCODEC} ${FFMPEG_LIBAVFILTER}
-                         ${FFMPEG_LIBAVFORMAT} ${FFMPEG_LIBAVUTIL}
-                         ${FFMPEG_LIBSWSCALE} ${FFMPEG_LIBSWRESAMPLE}
-                         ${FFMPEG_LIBPOSTPROC} ${FFMPEG_LDFLAGS})
-    list(APPEND FFMPEG_DEFINITIONS -DFFMPEG_VER_SHA=\"${FFMPEG_VERSION}\")
-
-    if(NOT TARGET ffmpeg)
-      add_library(ffmpeg ${FFMPEG_LIB_TYPE} IMPORTED)
-      set_target_properties(ffmpeg PROPERTIES
-                                   FOLDER "External Projects"
-                                   IMPORTED_LOCATION "${FFMPEG_LIBRARIES}"
-                                   INTERFACE_INCLUDE_DIRECTORIES "${FFMPEG_INCLUDE_DIRS}"
-                                   INTERFACE_LINK_LIBRARIES "${FFMPEG_LDFLAGS}"
-                                   INTERFACE_COMPILE_DEFINITIONS "${FFMPEG_DEFINITIONS}")
+    if(FFMPEG_FOUND)
+      set(FFMPEG_LDFLAGS ${PC_FFMPEG_LDFLAGS} CACHE STRING "ffmpeg linker flags")
+
+      set(FFMPEG_LIBRARIES ${FFMPEG_LIBAVCODEC} ${FFMPEG_LIBAVFILTER}
+                           ${FFMPEG_LIBAVFORMAT} ${FFMPEG_LIBAVUTIL}
+                           ${FFMPEG_LIBSWSCALE} ${FFMPEG_LIBSWRESAMPLE}
+                           ${FFMPEG_LIBPOSTPROC} ${FFMPEG_LDFLAGS})
+
+      # check if ffmpeg libs are statically linked
+      set(FFMPEG_LIB_TYPE SHARED)
+      foreach(_fflib IN LISTS FFMPEG_LIBRARIES)
+        if(${_fflib} MATCHES ".+\.a$" AND PC_FFMPEG_STATIC_LDFLAGS)
+          set(FFMPEG_LDFLAGS ${PC_FFMPEG_STATIC_LDFLAGS} CACHE STRING "ffmpeg linker flags" FORCE)
+          set(FFMPEG_LIB_TYPE STATIC)
+          break()
+        endif()
+      endforeach()
     endif()
-  endif()
-endif()
-
-# Internal FFMPEG
-if(NOT FFMPEG_FOUND)
-  include(ExternalProject)
-  include(cmake/scripts/common/ModuleHelpers.cmake)
-
-  get_archive_name(ffmpeg)
-
-  # allow user to override the download URL with a local tarball
-  # needed for offline build envs
-  if(FFMPEG_URL)
-    get_filename_component(FFMPEG_URL "${FFMPEG_URL}" ABSOLUTE)
   else()
```

Comment:
```suggestion
    elseif(NOT FFMPEG_PATH)
    message(STATUS "FFmpeg ${REQUIRED_FFMPEG_VERSION} not found, falling back to internal build")
    buildFFMPEG()
  else()
    message(FATAL_ERROR "FFmpeg not found, please consider using -DENABLE_INTERNAL_FFMPEG=ON")
  endif()

```

Reply:
hmm looks like I put this in the wrong location, but I guess you see the intent

Reply:
Have done a fixup. should be what your after. Didnt do an elseif, as i think its better to just add the ffmpeg_path check inside the else, and then either message, or internal build as commented.